core/src/services/routing.js

```diff
@@ -201,6 +201,14 @@ class RoutingClass {
         RoutingHelpers.setFeatureToggles(featureToggleProperty, path);
       }
 
+      // handle bookmarkable modal path
+      const additionalModalPath = RoutingHelpers.getModalPathFromPath();
+      if (additionalModalPath) {
+        const modalParams = RoutingHelpers.getModalParamsFromPath();
+        const { nodeObject } = await Navigation.extractDataFromPath(additionalModalPath);
+        LuigiNavigation.openAsModal(additionalModalPath, nodeObject.openNodeInModal || modalParams);
+      }
+
```

Comment:
async handleRouteChange(path, component, iframeElement, config, withoutSync)  function is getting pretty long and complex (it would be nice to refactor). 
Could you create a new function:

`  async handleBookmarkableModalPath(){
    const additionalModalPath = RoutingHelpers.getModalPathFromPath();
    if (additionalModalPath) {
      const modalParams = RoutingHelpers.getModalParamsFromPath();
      const { nodeObject } = await Navigation.extractDataFromPath(additionalModalPath);
      LuigiNavigation.openAsModal(additionalModalPath, nodeObject.openNodeInModal || modalParams);
    }
  }`


```suggestion
     await this.handleBookmarkableModalPath();
```

Reply:
Yep, I have not taken time to refactor yet, wanted to get the things working first. I appreciate your thoughts 👍